src/lib/HST/Effect/InputFile.hs

```diff
@@ -59,3 +61,14 @@ runWithInputFile initialFileMap = evalState initialFileMap . inputFileToState
           modify $ Map.insert filePath contents
           return contents
         Just contents -> return contents
+
+-- | Handles a computation by providing a map that maps file paths to contents.
+--   This handler does not use IO actions but returns the empty string if the
+--   given file was not found.
+runInputFileNoIO :: Map FilePath String -> Sem (InputFile ': r) a -> Sem r a
+runInputFileNoIO fileMap = runReader fileMap . reinterpret \case
+  GetInputFile filePath -> do
+    maybeContents <- asks $ Map.lookup filePath
+    case maybeContents of
+      Nothing       -> return ""
+      Just contents -> return contents
```

Comment:
Since the `runInputFile` handler throws an exception when the file does not exist, this handler should also fail instead of returning an empty string such that the tests are more representative.

```suggestion
-- | Handles a computation by providing a map that maps file paths to contents.
runInputFileNoIO :: Member Report r
                 => Map FilePath String
                 -> Sem (InputFile ': r) a
                 -> Sem r a
runInputFileNoIO fileMap = runReader fileMap . reinterpret \case
  GetInputFile filePath -> do
    maybeContents <- asks $ Map.lookup filePath
    case maybeContents of
      Nothing       ->
        reportFatal $ message Error S.Nothing $ "File not found: " ++ filePath
      Just contents -> return contents
```

You can then use [`setExpectationToIO . reportToSetExpectation`](https://github.com/FreeProving/haskell-src-transformations/blob/cdeb3fb7baea477e5c3575d242ff0a3cb0d9cbf9/src/test/HST/Effect/SetExpectation.hs) in the tests to handle the `Report` effect. Maybe define a helper function to your tests for handling all effects to avoid repetition.

Also note, that you don't need the reader effect. You can access the `fileMap` directly.

```suggestion
-- | Handles a computation by providing a map that maps file paths to contents.
runInputFileNoIO :: Member Report r
                 => Map FilePath String
                 -> Sem (InputFile ': r) a
                 -> Sem r a
runInputFileNoIO fileMap = interpret \case
  GetInputFile filePath -> case Map.lookup filePath fileMap of
    Nothing       ->
      reportFatal $ message Error S.Nothing $ "File not found: " ++ filePath
    Just contents -> return contents
```

Reply:
Unfortunately, I get a cyclic dependency:
HST.Util.Messages -> HST.Effect.InputFile -> HST.Effect.Report -> HST.Util.Messages

Reply:
Then we should either break the cyclic dependency with a `.hs-boot` file or at least throw an exception with `error`.